bin/incredible-squaring-avs/src/commands/avs/mod.rs

```diff
@@ -516,92 +679,289 @@ impl<Ext: clap::Args + fmt::Debug + Send + Sync + 'static> AvsCommand<Ext> {
+    }];
+    Ok(allocation_manager_instance
+        .modifyAllocations(signer.address(), allocate_params)
+        .send()
+        .await?
+        .get_receipt()
+        .await?
+        .transaction_hash)
+}
+
+/// registers the operator to the operator set by calling the allocation manager
+#[allow(clippy::too_many_arguments)]
+pub async fn register_for_operator_sets(
+    operator_set_id: u32,
+    bls_key_pair: BlsKeyPair,
+    registry_coordinator_address: Address,
+    allocation_manager: Address,
+    operator_pvt_key: Option<String>,
+    ecdsa_keystore_path: String,
+    ecdsa_keystore_password: String,
+    rpc_url: &str,
+    avs: Address,
+    socket: String,
+) -> eyre::Result<FixedBytes<32>> {
+    let signer;
+    if let Some(operator_key) = operator_pvt_key {
+        signer = PrivateKeySigner::from_str(&operator_key)?;
+    } else {
+        signer = LocalSigner::decrypt_keystore(ecdsa_keystore_path, ecdsa_keystore_password)?;
+    }
+    let s = signer.to_field_bytes();
+    let pvt_key = hex::encode(s).to_string();
+    let allocation_manager_instance =
+        AllocationManager::new(allocation_manager, get_signer(&pvt_key, rpc_url));
+
+    let contract_registry_coordinator =
+        RegistryCoordinator::new(registry_coordinator_address, get_provider(rpc_url));
+
+    let g1_hashed_msg_to_sign = contract_registry_coordinator
+        .pubkeyRegistrationMessageHash(signer.address())
+        .call()
+        .await
+        .map_err(|_| AvsRegistryError::PubKeyRegistrationMessageHash)?
+        ._0;
+
+    let sig = bls_key_pair
+        .sign_hashed_to_curve_message(alloy_g1_point_to_g1_affine(g1_hashed_msg_to_sign))
+        .g1_point();
+    let alloy_g1_point_signed_msg = convert_to_g1_point(sig.g1())?;
+    let g1_pub_key_bn254 = convert_to_g1_point(bls_key_pair.public_key().g1())?;
+    let g2_pub_key_bn254 = convert_to_g2_point(bls_key_pair.public_key_g2().g2())?;
+
+    let g2_point_x: Vec<DynSolValue> = vec![
+        DynSolValue::Uint(g2_pub_key_bn254.X[0], 256),
+        DynSolValue::Uint(g2_pub_key_bn254.X[1], 256),
+    ];
+    let g2_point_y: Vec<DynSolValue> = vec![
+        DynSolValue::Uint(g2_pub_key_bn254.Y[0], 256),
+        DynSolValue::Uint(g2_pub_key_bn254.Y[1], 256),
+    ];
+    let encoded_params_with_socket = DynSolValue::Tuple(vec![
+        DynSolValue::String(socket),
+        DynSolValue::Uint(alloy_g1_point_signed_msg.X, 256),
+        DynSolValue::Uint(alloy_g1_point_signed_msg.Y, 256),
+        DynSolValue::Uint(g1_pub_key_bn254.X, 256),
+        DynSolValue::Uint(g1_pub_key_bn254.Y, 256),
+        DynSolValue::FixedArray(g2_point_x),
+        DynSolValue::FixedArray(g2_point_y),
+    ])
+    .abi_encode_params();
+    let register_params = RegisterParams {
+        avs,
+        operatorSetIds: vec![operator_set_id],
+        data: encoded_params_with_socket.into(),
+    };
+    Ok(allocation_manager_instance
+        .registerForOperatorSets(signer.address(), register_params)
```

Comment:
Shouldn't we use the `ELChainWriter::register_for_operator_sets` here?

Reply:
done